pallets/ds-accounts/src/lib.rs

```diff
@@ -209,6 +228,29 @@ decl_module! {
             Ok(())
         }
 
+        #[weight = <T as Trait>::WeightInfo::register_pilot()]
+        pub fn register_pilot(origin, account: T::AccountId) -> dispatch::DispatchResult {
+            // Check that the extrinsic was signed and get the signer.
+            // This function will return an error if the extrinsic is not signed.
+            // https://substrate.dev/docs/en/knowledgebase/runtime/origin
+            let who = ensure_signed(origin)?;
+            ensure!(Self::account_is_registrar(&who), Error::<T>::NotAuthorized);
+            ensure!(Self::account_is_none_role(&account), Error::<T>::NotAllowedRole);
```

Comment:
Every read from persistent storage is very expensive operation and we don't have to read the save value twice. 
It's better to  eliminate redundant  Store read  here and combine  check with mutate method call.